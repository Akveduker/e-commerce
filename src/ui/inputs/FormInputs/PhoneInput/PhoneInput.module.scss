@import "../../../../styles/_mixins.scss";
.container {
    position: relative;
}
.input {
    @include formInput;
}
.name {
    @include authorizationLabelText;
}
