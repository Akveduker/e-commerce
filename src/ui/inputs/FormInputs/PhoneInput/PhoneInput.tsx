import { formatePhoneInInput } from '../../../../utils/filters/formatePhoneInInput/formatePhoneInInput';
import { FormDefaultInput } from '../../../../types/formInput/formInput';
import { useInputsStateChangers } from '../../../../hooks/form/useInputsStateChangers';
import InputWithValidators from '../parts/InputWithValidators/InputWithValidators';
const PhoneInput = <T extends string[]>(props: FormDefaultInput<T>) => {
    const { dataItemState, inputTitle, errorTitle, dispatch, stateItemKey, ...rest } = props
    const { status, value, errorBody } = dataItemState
    const [onChange, onBlur] = useInputsStateChangers(props, formatePhoneInInput(value))
    return (
        <InputWithValidators
            inputType='mask'
            mask={"+7 999 999 99 99"}
            inputTitle={inputTitle}
            errorTitle={errorTitle}
            status={status}
            errorBody={errorBody}
            onChange={onChange}
            onBlur={onBlur}
            type='text'
            value={value}
            {...rest}
        />
    );
};

export default PhoneInput;