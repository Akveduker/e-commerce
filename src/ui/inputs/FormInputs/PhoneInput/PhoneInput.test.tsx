import { render, screen } from "@testing-library/react"
import PhoneInput from "./PhoneInput"

describe('PhoneInput', () => {
    test('Только инпут с маской,снапшот', () => {
        const { container } = render(
            <PhoneInput
                inputTitle="title"
                errorTitle="error"
                stateItemKey="phone"
                dispatch={jest.fn}
                dataItemState={{ errorBody: '', status: 'idle', value: '', validators: [] }}
            />
        )
        expect(screen.queryByTestId('default')).toBeNull()
        expect(container).toMatchSnapshot()
    })
})