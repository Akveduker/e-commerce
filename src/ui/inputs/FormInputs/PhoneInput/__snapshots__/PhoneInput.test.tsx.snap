// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInput Только инпут с маской,снапшот 1`] = `
<div>
  <label>
    <p
      class="name"
    >
      title
    </p>
    <label
      class="with-children container"
      data-testid="label"
    >
      <input
        class="with-children__input"
        type="text"
        value=""
      />
      <div
        class="with-children__children"
      >
        <div />
      </div>
    </label>
    <div />
  </label>
</div>
`;
